crates/ide-diagnostics/src/handlers/remove_unnecessary_else.rs

```diff
@@ -41,9 +41,11 @@ fn fixes(ctx: &DiagnosticsContext<'_>, d: &RemoveUnnecessaryElse) -> Option<Vec<
         indent = indent + 1;
     }
     let else_replacement = match if_expr.else_branch()? {
-        ast::ElseBranch::Block(ref block) => {
-            block.statements().map(|stmt| format!("\n{indent}{stmt}")).join("")
-        }
+        ast::ElseBranch::Block(ref block) => block
+            .statements()
+            .map(|stmt| format!("\n{indent}{stmt}"))
+            .chain(block.tail_expr().map(|tail| format!("\n{indent}{tail}")))
+            .join(""),
         ast::ElseBranch::IfExpr(ref nested_if_expr) => {
```

Comment:
nit: same here, you can remove this unnecessary ref, again my fault not yours 😅.